lib/collection/src/shards/transfer/shard_transfer.rs

```diff
@@ -282,25 +283,23 @@ async fn transfer_snapshot(
         )));
     };
 
-    error_if_cancelled(&cancel)?;
-
     // Queue proxy local shard
     replica_set
         .queue_proxify_local(remote_shard.clone())
         .await?;
+
     debug_assert!(
         replica_set.is_queue_proxy().await,
         "Local shard must be a queue proxy"
     );
 
-    error_if_cancelled(&cancel)?;
-
     // Create shard snapshot
     log::trace!("Creating snapshot of shard {shard_id} for shard snapshot transfer");
     let snapshot_description = shard_holder_read
         .create_shard_snapshot(snapshots_path, collection_name, shard_id, temp_dir)
         .await?;
 
+    // TODO: If future is cancelled until `get_shard_snapshot_path` resolves, shard snapshot may not be cleaned up...
```

Comment:
Maybe we do want some function that returns a `TempFile` to prevent this.

Reply:
Another refactoring of `create_shard_snapshot` (or adding a separate method) 🥲